XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Simulation/Hands/SimulatedHandControllerPose.cs

```diff
@@ -0,0 +1,289 @@
+                Quaternion r = localJointPoses[i].Rotation;
+
+                // Pose offset are for right hand, mirror on X axis if left hand is needed
+                if (handedness == Handedness.Left)
+                {
+                    p.x = -p.x;
+                    r.y = -r.y;
+                    r.z = -r.z;
+                }
+
+                // Apply camera transform
+                p = cameraRotation * p;
+                r = cameraRotation * r;
+
+                // Apply external transform
+                p = position + rotation * p;
+                r = rotation * r;
+
+                jointsOut[i] = new MixedRealityPose(p, r);
+            }
+        }
+
+        /// <summary>
+        /// Takes world space joint poses from any hand and convert into right-hand, camera-space poses.
+        /// </summary>
+        public void ParseFromJointPoses(MixedRealityPose[] joints, Handedness handedness, Quaternion rotation, Vector3 position)
+        {
+            var invRotation = Quaternion.Inverse(rotation);
+            var invCameraRotation = Quaternion.Inverse(MixedRealityToolkit.CameraSystem.CameraRig.PlayerCamera.transform.rotation);
+
+            for (int i = 0; i < jointCount; i++)
+            {
+                Vector3 p = joints[i].Position;
+                Quaternion r = joints[i].Rotation;
+
+                // Apply inverse external transform
+                p = invRotation * (p - position);
+                r = invRotation * r;
+
+                // To camera space
+                p = invCameraRotation * p;
+                r = invCameraRotation * r;
+
+                // Pose offset are for right hand, mirror on X axis if left hand is given
+                if (handedness == Handedness.Left)
+                {
+                    p.x = -p.x;
+                    r.y = -r.y;
+                    r.z = -r.z;
+                }
+
+                localJointPoses[i] = new MixedRealityPose(p, r);
+            }
+        }
+
+        /// <summary>
+        /// Set all poses to zero.
+        /// </summary>
+        public void SetZero()
+        {
+            for (int i = 0; i < jointCount; i++)
+            {
+                localJointPoses[i] = MixedRealityPose.ZeroIdentity;
+            }
+        }
+
+        /// <summary>
+        /// Copy data from another simulated hand pose.
+        /// </summary>
+        public void Copy(SimulatedHandControllerPose other)
+        {
+            Array.Copy(other.localJointPoses, localJointPoses, jointCount);
+        }
+
+        /// <summary>
+        /// Interpolates between the poses a and b by the interpolant t.
+        /// </summary>
+        /// <param name="a">Pose at t = 0.</param>
+        /// <param name="b">Pose at t = 1.</param>
+        /// <param name="t">The parameter t is clamped to the range [0,1].</param>
+        public static SimulatedHandControllerPose Lerp(SimulatedHandControllerPose a, SimulatedHandControllerPose b, float t)
```

Comment:
Consider making an extension

Reply:
I disagree here. Check how Untiy's Vector3 Lerp is implemented. There is no aVector.Lerp there as well which makes sense IMO, because you are not modifying any of the poses but instead creating a new one that is somwehere in between a and b. Don't see an extension fit here.